tests/unit/test_charm.py

```diff
@@ -0,0 +1,178 @@
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+# Learn more about testing at: https://juju.is/docs/sdk/testing
+
+import json
+import unittest
+
+from ops.model import ActiveStatus, BlockedStatus
+from ops.testing import Harness
+
+from charm import PrometheusScrapeTargetCharm
+
+
+class TestCharm(unittest.TestCase):
+    def setUp(self):
+        """Flake8 forces me to write meaningless docstrings."""
+        self.harness = Harness(PrometheusScrapeTargetCharm)
+        self.harness.set_model_info(name="lma", uuid="e40bf1a0-91f4-45a5-9f35-eb30fd010e4d")
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+
+    def test_no_data(self):
+        """Test charm is blocked when no configs are provided."""
+        self.harness.set_leader(True)
+
+        downstream_rel_id = self.harness.add_relation("metrics-endpoint", "prometheus-k8s")
+
+        self.assertEqual(
+            {},
+            self.harness.get_relation_data(downstream_rel_id, self.harness.charm.app.name),
+        )
+
+        self.assertEqual(self.harness.model.unit.status, BlockedStatus("No targets specified"))
+
+    def test_no_leader(self):
```

Comment:
Same point regarding test name.

Reply:
this test is not very useful as-is because a non-leader cannot change app data anyway. Maybe replace with `self.assertRaises(...)`?